txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -250,5 +266,9 @@ rollbackTxUtxo
 rollbackTxUtxo (txAux, undo) = do
     let tx@UnsafeTx {..} = taTx txAux
     let txid = hash tx
-    mapM_ utxoDel $ take (length _txOutputs) $ map (TxIn txid) [0..]
-    mapM_ (uncurry utxoPut) $ NE.zip _txInputs undo
+    mapM_ utxoDel $ take (length _txOutputs) $ map (TxInUtxo txid) [0..]
+    mapM_ (uncurry utxoPut) $ concatMap knownInputAndUndo $ toList $ NE.zip _txInputs undo
+  where
+    knownInputAndUndo (_, Nothing)         = []
+    knownInputAndUndo (TxInUnknown _ _, _) = []
+    knownInputAndUndo (inp, Just u)        = [(inp, u)]
```

Comment:
To be honest, returning `Maybe` and using `mapMaybe` looks more conventional to me, but it's not important.
I think it would good to also yell loudly if we encounter `Nothing` and not `TxInUnknown` or `Just` and `TxInUnknown`.